packages/extension-vscode/src/server.ts

```diff
@@ -109,7 +106,7 @@ const loadModule = async <T>(context: string, name: string): Promise<T | null> =
 };
 
 // Load a user configuration, falling back to 'development' if none exists.
-const loadUserConfig = (directory: string, Configuration: typeof config.Configuration): UserConfig => {
+const loadUserConfig = (directory: string, Configuration: typeof hint.Configuration): UserConfig => {
```

Comment:
Feels like some version of this could be provided by the public `hint` API... perhaps some more discussion is needed here to figure out how to best share more code.

Reply:
After discussing there's going to be a new static method that will try to load a user config. It will accept an starting path and will return `null` if nothing is found or works.